rad/src/lib.rs

```diff
@@ -32,13 +32,17 @@ pub fn run_retrieval_with_data(retrieve: &RADRetrieve, response: String) -> Resu
 }
 
 /// Run retrieval stage of a data request.
-pub fn run_retrieval(retrieve: &RADRetrieve) -> Result<RadonTypes> {
+pub async fn run_retrieval(retrieve: &RADRetrieve) -> Result<RadonTypes> {
     match retrieve.kind {
         RADType::HttpGet => {
-            let response = reqwest::get(&retrieve.url)
-                .map_err(RadError::from)?
+            let response = reqwest::Client::builder()
+                .build()?
+                .get(&retrieve.url)
+                .send()
+                .await?
+                // TODO: replace with .bytes() and let RADON handle the encoding?
```

Comment:
The `reqwest` crate supports many custom encodings by respecting the `Content-Type` HTTP header. It also supports a default encoding with `text_with_charset`, although the encoding specified in the header has priority.

We can change `.text()` to `.bytes()` and let RADON do the decoding, but we lose the information from the HTTP headers.